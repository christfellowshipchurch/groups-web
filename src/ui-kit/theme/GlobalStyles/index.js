import { createGlobalStyle } from 'styled-components';
import { normalize } from 'styled-normalize';

import { h2Style } from '../../typography/H2';
import { h3Style } from '../../typography/H3';
import { h4Style } from '../../typography/H4';
import { h5Style } from '../../typography/H5';
import { h6Style } from '../../typography/H6';

const GlobalStyles = createGlobalStyle`
  ${normalize}

  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }

  html {
    color: ${({ theme }) => theme.colors.text.primary};
    font-size: 100%;
    font-family: ${({ theme }) => theme.typography.fontFamily};
  }

  ::selection {
    background: ${({ theme }) => theme.colors.primary}
    text-shadow: none;
  }

  /* Individal typography styles are imported to and referenced from the global stylesheet. See
     typogrpahy components (/ui-kit/typography) for style definitions */
  ${h2Style}
  ${h3Style}
  ${h4Style}
  ${h5Style}
  ${h6Style}

  /* Copied from https://github.com/buildo/react-placeholder/blob/master/src/reactPlaceholder.scss
  to reduce technical complexity with importing css */
  .show-loading-animation {
    &.rect-shape,
    &.round-shape,
    &.text-row,
    &.custom-placeholder, /* We add a custom class for use in our placeholder components */
    .rect-shape,
    .round-shape,
    .text-row,
    .custom-placeholder { /* We add a custom class for use in our placeholder components */
      animation: react-placeholder-pulse 1.5s infinite;
    }
  }

  @keyframes react-placeholder-pulse {
    0% {
      opacity: .6;
    }
    50% {
      opacity: 1;
    }
    100% {
      opacity: .6;
    }
  }
`;

export default GlobalStyles;
