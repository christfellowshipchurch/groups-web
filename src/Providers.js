import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
