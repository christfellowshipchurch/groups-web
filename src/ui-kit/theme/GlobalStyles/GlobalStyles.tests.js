import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../../../Providers';

import GlobalStyles from '.';

describe.only('the GlobalStyles component', () => {
  it('should render', () => {
    renderer.create(
      <Providers>
        <GlobalStyles />
      </Providers>
    );
    const styleTags = document.head.getElementsByTagName('style');
    expect(styleTags).toMatchSnapshot();
  });
});
