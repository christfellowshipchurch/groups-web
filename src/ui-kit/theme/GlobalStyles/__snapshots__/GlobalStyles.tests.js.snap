// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the GlobalStyles component should render 1`] = `
HTMLCollection [
  <style
    data-styled="active"
    data-styled-version="5.1.1"
  >
    html{line-height:1.15;-webkit-text-size-adjust:100%;}
    body{margin:0;}
    main{display:block;}
    h1{font-size:2em;margin:0.67em 0;}
    hr{box-sizing:content-box;height:0;overflow:visible;}
    pre{font-family:monospace,monospace;font-size:1em;}
    a{background-color:transparent;}
    abbr[title]{border-bottom:none;-webkit-text-decoration:underline;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted;}
    b,strong{font-weight:bolder;}
    code,kbd,samp{font-family:monospace,monospace;font-size:1em;}
    small{font-size:80%;}
    sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline;}
    sub{bottom:-0.25em;}
    sup{top:-0.5em;}
    img{border-style:none;}
    button,input,optgroup,select,textarea{font-family:inherit;font-size:100%;line-height:1.15;margin:0;}
    button,input{overflow:visible;}
    button,select{text-transform:none;}
    button,[type="button"],[type="reset"],[type="submit"]{-webkit-appearance:button;}
    button::-moz-focus-inner,[type="button"]::-moz-focus-inner,[type="reset"]::-moz-focus-inner,[type="submit"]::-moz-focus-inner{border-style:none;padding:0;}
    button:-moz-focusring,[type="button"]:-moz-focusring,[type="reset"]:-moz-focusring,[type="submit"]:-moz-focusring{outline:1px dotted ButtonText;}
    fieldset{padding:0.35em 0.75em 0.625em;}
    legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal;}
    progress{vertical-align:baseline;}
    textarea{overflow:auto;}
    [type="checkbox"],[type="radio"]{box-sizing:border-box;padding:0;}
    [type="number"]::-webkit-inner-spin-button,[type="number"]::-webkit-outer-spin-button{height:auto;}
    [type="search"]{-webkit-appearance:textfield;outline-offset:-2px;}
    [type="search"]::-webkit-search-decoration{-webkit-appearance:none;}
    ::-webkit-file-upload-button{-webkit-appearance:button;font:inherit;}
    details{display:block;}
    summary{display:list-item;}
    template{display:none;}
    [hidden]{display:none;}
    h1,.H1{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:2.6875rem;font-weight:900;line-height:3.0906249999999997rem;}
    h1.custom-placeholder,.H1.custom-placeholder{height:2.6875rem;}
    h2,.H2{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:2.25rem;font-weight:900;line-height:2.5875rem;}
    h2.custom-placeholder,.H2.custom-placeholder{height:2.25rem;}
    h3,.H3{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:1.5rem;font-weight:900;line-height:1.7249999999999999rem;}
    h3.custom-placeholder,.H3.custom-placeholder{height:1.5rem;}
    h4,.H4{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:1rem;font-weight:700;line-height:1.5rem;}
    h4.custom-placeholder,.H4.custom-placeholder{height:1rem;}
    h5,.H5{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:0.875rem;font-weight:500;line-height:1.3125rem;}
    h5.custom-placeholder,.H5.custom-placeholder{height:0.875rem;}
    h6,.H6{color:#505050;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:0.75rem;font-weight:500;line-height:1.125rem;}
    h6.custom-placeholder,.H6.custom-placeholder{height:0.75rem;}
    .show-loading-animation.rect-shape,.show-loading-animation.round-shape,.show-loading-animation.text-row,.show-loading-animation.custom-placeholder,.show-loading-animation .rect-shape,.show-loading-animation .round-shape,.show-loading-animation .text-row,.show-loading-animation .custom-placeholder{-webkit-animation:react-placeholder-pulse 1.5s infinite;animation:react-placeholder-pulse 1.5s infinite;}
    @-webkit-keyframes react-placeholder-pulse{0%{opacity:.6;}50%{opacity:1;}100%{opacity:.6;}}
    @keyframes react-placeholder-pulse{0%{opacity:.6;}50%{opacity:1;}100%{opacity:.6;}}
    html{line-height:1.15;-webkit-text-size-adjust:100%;}
    body{margin:0;}
    main{display:block;}
    h1{font-size:2em;margin:0.67em 0;}
    hr{box-sizing:content-box;height:0;overflow:visible;}
    pre{font-family:monospace,monospace;font-size:1em;}
    a{background-color:transparent;}
    abbr[title]{border-bottom:none;-webkit-text-decoration:underline;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted;}
    b,strong{font-weight:bolder;}
    code,kbd,samp{font-family:monospace,monospace;font-size:1em;}
    small{font-size:80%;}
    sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline;}
    sub{bottom:-0.25em;}
    sup{top:-0.5em;}
    img{border-style:none;}
    button,input,optgroup,select,textarea{font-family:inherit;font-size:100%;line-height:1.15;margin:0;}
    button,input{overflow:visible;}
    button,select{text-transform:none;}
    button,[type="button"],[type="reset"],[type="submit"]{-webkit-appearance:button;}
    button::-moz-focus-inner,[type="button"]::-moz-focus-inner,[type="reset"]::-moz-focus-inner,[type="submit"]::-moz-focus-inner{border-style:none;padding:0;}
    button:-moz-focusring,[type="button"]:-moz-focusring,[type="reset"]:-moz-focusring,[type="submit"]:-moz-focusring{outline:1px dotted ButtonText;}
    fieldset{padding:0.35em 0.75em 0.625em;}
    legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal;}
    progress{vertical-align:baseline;}
    textarea{overflow:auto;}
    [type="checkbox"],[type="radio"]{box-sizing:border-box;padding:0;}
    [type="number"]::-webkit-inner-spin-button,[type="number"]::-webkit-outer-spin-button{height:auto;}
    [type="search"]{-webkit-appearance:textfield;outline-offset:-2px;}
    [type="search"]::-webkit-search-decoration{-webkit-appearance:none;}
    ::-webkit-file-upload-button{-webkit-appearance:button;font:inherit;}
    details{display:block;}
    summary{display:list-item;}
    template{display:none;}
    [hidden]{display:none;}
    h1,.H1{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:2.6875rem;font-weight:900;line-height:3.0906249999999997rem;}
    h1.custom-placeholder,.H1.custom-placeholder{height:2.6875rem;}
    h2,.H2{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:2.25rem;font-weight:900;line-height:2.5875rem;}
    h2.custom-placeholder,.H2.custom-placeholder{height:2.25rem;}
    h3,.H3{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:1.5rem;font-weight:900;line-height:1.7249999999999999rem;}
    h3.custom-placeholder,.H3.custom-placeholder{height:1.5rem;}
    h4,.H4{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:1rem;font-weight:700;line-height:1.5rem;}
    h4.custom-placeholder,.H4.custom-placeholder{height:1rem;}
    h5,.H5{color:#303030;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:0.875rem;font-weight:500;line-height:1.3125rem;}
    h5.custom-placeholder,.H5.custom-placeholder{height:0.875rem;}
    h6,.H6{color:#505050;font-family:Inter,-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;font-size:0.75rem;font-weight:500;line-height:1.125rem;}
    h6.custom-placeholder,.H6.custom-placeholder{height:0.75rem;}
    .show-loading-animation.rect-shape,.show-loading-animation.round-shape,.show-loading-animation.text-row,.show-loading-animation.custom-placeholder,.show-loading-animation .rect-shape,.show-loading-animation .round-shape,.show-loading-animation .text-row,.show-loading-animation .custom-placeholder{-webkit-animation:react-placeholder-pulse 1.5s infinite;animation:react-placeholder-pulse 1.5s infinite;}
    @-webkit-keyframes react-placeholder-pulse{0%{opacity:.6;}50%{opacity:1;}100%{opacity:.6;}}
    @keyframes react-placeholder-pulse{0%{opacity:.6;}50%{opacity:1;}100%{opacity:.6;}}
  </style>,
]
`;
